Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Driver.hs

```diff
@@ -1388,44 +1399,25 @@ getDriverPayments (personId, merchantId_) mbFrom mbTo mbStatus mbLimit mbOffset
-  }
-  deriving (Generic, ToJSON, ToSchema, FromJSON)
-
-data DriverDuesEntity = DriverDuesEntity
-  { date :: UTCTime,
-    amount :: Money,
-    earnings :: Money,
-    offers :: [OfferEntity]
-  }
-  deriving (Generic, ToJSON, ToSchema, FromJSON)
-
-data OfferEntity = OfferEntity
-  { title :: Maybe Text,
-    description :: Maybe Text,
-    tnc :: Maybe Text
-  }
-  deriving (Generic, ToJSON, ToSchema, FromJSON)
-
-getDriverDues :: (EsqDBReplicaFlow m r, EsqDBFlow m r, EncFlow m r, CacheFlow m r) => (Id SP.Person, Id DM.Merchant) -> m DriverDuesResp
-getDriverDues (personId, _merchantId) = do
-  driverPlan <- runInReplica $ QDPlan.findByDriverId personId >>= fromMaybeM (NoCurrentPlanForDriver personId.getId)
-  plan <- QPD.findByIdAndPaymentMode driverPlan.planId driverPlan.planType >>= fromMaybeM (PlanNotFound driverPlan.planId.getId)
-  dueInvoices <- runInReplica $ QDF.findAllPendingAndDueDriverFeeByDriverId personId
-  return $
-    DriverDuesResp
-      { dues = buildDriverDuesEntity <$> dueInvoices,
-        totalDue = sum $ map (\dueInvoice -> fromIntegral dueInvoice.govtCharges + fromIntegral dueInvoice.platformFee.fee + dueInvoice.platformFee.cgst + dueInvoice.platformFee.sgst) dueInvoices,
-        overdueThreshold = plan.maxCreditLimit
-      }
+clearDriverDues :: (EsqDBReplicaFlow m r, EsqDBFlow m r, EncFlow m r, CacheFlow m r) => (Id SP.Person, Id DM.Merchant) -> m ClearDuesRes
+clearDriverDues (personId, _merchantId) = do
+  driverPlan <- QDPlan.findByDriverId personId >>= fromMaybeM (NoCurrentPlanForDriver personId.getId)
+  void $ QPD.findByIdAndPaymentMode driverPlan.planId driverPlan.planType >>= fromMaybeM (PlanNotFound driverPlan.planId.getId)
+  dueDriverFees <- QDF.findAllPendingAndDueDriverFeeByDriverId personId
+  invoices <- groupBy (\inv1 inv2 -> getId (inv1.id) == getId (inv2.id)) . sortBy (\inv1 inv2 -> compare (getId $ inv1.id) (getId $ inv2.id)) <$> QINV.findAllByDriverFeeIdAndStatus (dueDriverFees <&> (.id)) INV.ACTIVE_INVOICE
+  let invoice = validateExistingInvoice (listToMaybe invoices) dueDriverFees
+  if null invoices || (length invoices == 1 && isJust invoice)
+    then do
+      when (isJust invoice) $ QDF.updateFeeTypeByIds DDF.RECURRING_INVOICE (dueDriverFees <&> (.id)) =<< getCurrentTime
+      mkClearDuesResp <$> SPayment.createOrder (personId, _merchantId) dueDriverFees Nothing invoice
+    else throwError OngoingPaymentExecution
   where
-    buildDriverDuesEntity DDF.DriverFee {..} =
-      DriverDuesEntity
-        { date = createdAt,
-          amount = round $ fromIntegral govtCharges + fromIntegral platformFee.fee + platformFee.cgst + platformFee.sgst,
-          earnings = totalEarnings,
-          offers = []
-        }
+    validateExistingInvoice invoicesGroup driverFees = do
+      case invoicesGroup of
+        Just invoices -> do
+          if length ((invoices <&> getId . (.driverFeeId)) `intersect` (driverFees <&> getId . (.id))) == length invoices
```

Comment:
Why is this check needed? You are already fetching invoices specific to these driverFee IDs